### ISSUE TITLE
Minimizing Squared Distance

### ISSUE THREAD

**Issue by photopresentr** [markmfredrickson/optmatch#135]
I have treatment/controls distinguished by geographic location. The matching strategy I would like to use minimize *squared* Euclidean distance. However, the ``optmatch`` package by default minimizes absolute distance. 

In my use case, this means I get one observation with a great match and one observation with a terrible match.

Minimal example:

```
df <- rbind(
  data.frame(
    treat = 0,
    pos = 1:4
  ),
  data.frame(
    treat = 1,
    pos= 1:4 + 10
  )
)

dst <- match_on(treat ~ pos, data = df, method = "euclidean")
df$group <- fullmatch(dst, data = df,  remove.unmatchables = T);
df
```

Result:

```
  treat pos group
1     0   1   1.4 - terrible match: distance 13
2     0   2   1.3
3     0   3   1.2
4     0   4   1.1 - great match: distance 7
5     1  11   1.1
6     1  12   1.2
7     1  13   1.3
8     1  14   1.4
```

In an ideal world, I'd like to have distance 10 for everybody (in this example), i.e. minimize square distance.

# Workaround: Matching by Squared Distance

[Create  a custom distance function](https://github.com/markmfredrickson/optmatch/wiki/Implementing-non-standard-matching-distances) that returns the squared Euclidean distance, and then match on that:

```
compute_squaredistance <- function(index, data, z) {
  apply(index, 1, function(tcpair) {
    sum(abs(data[tcpair[1], ] - data[tcpair[2], ]))**2
  })
}
mydist <- match_on(treat ~ pos, data = df, method = compute_squaredistance)
```
This works, but it is not very readable

**Comment by benthestatistician** [comment]
Thanks for this comment, @photopresentr . 

To building on your minimal example (thanks), you can also get what you're after with

```{r}
df$group2 <- fullmatch(dst^2, data = df)
```

We should probably update [Create a custom distance function](https://github.com/markmfredrickson/optmatch/wiki/Implementing-non-standard-matching-distances) wiki to point users in this direction. I may change that to be the mission of this GH issue.

I notice that you've set `remove.unmatchables=T` in your example. That's a pairmatch-specific argument, as it happens; `fullmatch` always does the equivalent thing. I've been thinking that we should change the pairmatch default to be `remove.unmatchables=T` rather than `F`, making the removal of unmatchables be the general optmatch default.

**Comment by photopresentr** [comment]
This is a much better solution @benthestatistician , thanks for this!

I don't want to hijack this issue, but another issue I ran into - purely my fault too - was that I spelled "caliper" as "calipher" and was wondering why it did not work. Perhaps throwing an error when arguments are not recognised could be also a valuable addition to this great library.

Thanks for all the work and very useful software!
